frontend/src/routes/Governance/common/util.tsx

```diff
@@ -650,18 +651,81 @@ export function hasInformOnlyPolicies(items: Array<PolicyTableItem>): boolean {
   return true
 }
 
+// type guard
+function isPolicy(object: Policy | DiscoverdPolicyItem): object is Policy {
+  return object && (object as Policy).metadata !== undefined
+}
+
 export function getPolicySource(
-  policy: Policy,
+  policy: Policy | DiscoverdPolicyItem,
   helmReleases: HelmRelease[],
   channels: Channel[],
   subscriptions: Subscription[],
   t: TFunction<string, undefined>
 ): string | JSX.Element {
-  const isExternal = resolveExternalStatus(policy)
+  if (!policy) {
+    // Prevent error in detail page
+    return <></>
+  }
+
+  const isExternal = policy && isPolicy(policy) ? resolveExternalStatus(policy) : policy._isExternal || false
   let source: string | JSX.Element = t('Local')
   if (isExternal) {
-    const policySource = resolveSource(policy, helmReleases, channels, subscriptions)
+    const policySource = isPolicy(policy)
+      ? resolveSource(policy.metadata.annotations ?? {}, helmReleases, channels, subscriptions)
+      : resolveSource(policy.annotations ?? {}, helmReleases, channels, subscriptions)
     source = policySource ? getSource(policySource, isExternal, t) : t('Managed externally')
   }
   return source
 }
+
+export function getEngineString(kind: string): string {
+  switch (kind) {
+    case 'ConfigurationPolicy':
+    case 'CertificatePolicy':
+    case 'OperatorPolicy':
+      return 'Open Cluster Management'
+    default:
+      return 'Unknown'
+  }
+}
+
+export function getEngineWithSvg(kind: string): JSX.Element {
+  switch (kind) {
+    case 'ConfigurationPolicy':
+    case 'CertificatePolicy':
+    case 'OperatorPolicy':
+      return (
+        <div style={{ display: 'flex', alignItems: 'center', gap: 8 }}>
+          <img
+            src="https://open-cluster-management.io/ocm.svg"
+            alt="ocm"
+            style={{
+              height: 24,
+              width: 25,
+            }}
+          />{' '}
+          Open Cluster Management
+        </div>
+      )
+    default:
+      return <>Unknown</>
+  }
+}
+
+// ex: The label looks like this: this "app=demo; open-cluster-management.io/policy-set=cluster-hardening",
+export function getStringLabelValue(labels: string, labelKey: string): string {
+  if (!labels) {
+    return ''
+  }
+
+  const parsed = labels.split('; ')
+  return parsed.filter((p: string) => p.includes(labelKey))?.[0]?.split('=')[1]
+}
+
+export function parseDiscoveryPolicies(data: any): any {
+  return JSON.parse(JSON.stringify(data), (_, v: string) => {
+    if (v === 'true') return true
```

Comment:
When is this necessary? I'm just curious if we have an example that we can put in a comment here.

Reply:
From backend boolean come as string so we need to parse properly